torchvision/csrc/cpu/video_reader/FfmpegStream.cpp

```diff
@@ -0,0 +1,267 @@
+>>>>>>> 73af0db8f428e9137ee85db70a1532a96ca407d5
+=======
+void FfmpegStream::flush(int getPtsOnly, DecoderOutput& decoderOutput)
+{
+  VLOG(1) << "Media Type: " << getMediaType() << ", flush stream.";
+>>>>>>> Stashed changes
+  // need to receive frames before entering draining mode
+  receiveAvailFrames(getPtsOnly, decoderOutput);
+
+  VLOG(2) << "send nullptr packet";
+  sendPacket(nullptr);
+  // receive remaining frames after entering draining mode
+  receiveAvailFrames(getPtsOnly, decoderOutput);
+
+  avcodec_flush_buffers(codecCtx_);
+}
+
+bool FfmpegStream::isFramePtsInRange() {
+  CHECK(frame_);
+  auto pts = frame_->pts;
+  auto startPts = this->getStartPts();
+  auto endPts = this->getEndPts();
+<<<<<<< Updated upstream
+<<<<<<< HEAD
+  VLOG(2) << "isPtsInRange. pts: " << pts << ", startPts: " << startPts
+          << ", endPts: " << endPts;
+  return (pts == AV_NOPTS_VALUE) ||
+      (pts >= startPts && (endPts > 0 ? pts <= endPts : true));
+=======
+  VLOG(2) << "isPtsInRange. pts: " << pts
+          << ", startPts: " << startPts
+          << ", endPts: " << endPts;
+  return (pts == AV_NOPTS_VALUE) || (pts >= startPts && (endPts >= 0 ? pts <= endPts : true));
+>>>>>>> 73af0db8f428e9137ee85db70a1532a96ca407d5
+=======
+  VLOG(2) << "isPtsInRange. pts: " << pts
+          << ", startPts: " << startPts
+          << ", endPts: " << endPts;
+  return (pts == AV_NOPTS_VALUE) || (pts >= startPts && (endPts >= 0 ? pts <= endPts : true));
+>>>>>>> Stashed changes
+}
+
+bool FfmpegStream::isFramePtsExceedRange() {
+  if (frame_) {
+    auto endPts = this->getEndPts();
+<<<<<<< Updated upstream
+<<<<<<< HEAD
+    VLOG(2) << "isFramePtsExceedRange. last_pts_: " << last_pts_
+            << ", endPts: " << endPts;
+    return endPts > 0 ? last_pts_ >= endPts : false;
+=======
+    VLOG(2) << "isFramePtsExceedRange. last_pts_: " << last_pts_ << ", endPts: " << endPts;
+    return endPts >= 0 ? last_pts_ >= endPts : false;
+>>>>>>> 73af0db8f428e9137ee85db70a1532a96ca407d5
+=======
+    VLOG(2) << "isFramePtsExceedRange. last_pts_: " << last_pts_ << ", endPts: " << endPts;
+    return endPts >= 0 ? last_pts_ >= endPts : false;
+>>>>>>> Stashed changes
+  } else {
+    return true;
+  }
+}
+
+// seek a frame
+int FfmpegStream::seekFrame(int64_t seekPts) {
+  // translate margin from second to pts
+  int64_t margin = (int64_t)(
+      seekFrameMargin_ * (double)inputCtx_->streams[index_]->time_base.den /
+      (double)inputCtx_->streams[index_]->time_base.num);
+  int64_t real_seekPts = (seekPts - margin) > 0 ? (seekPts - margin) : 0;
+  VLOG(2) << "seek margin: " << margin;
+  VLOG(2) << "real seekPts: " << real_seekPts;
+  int ret = av_seek_frame(
+      inputCtx_,
+      index_,
+      (seekPts - margin) > 0 ? (seekPts - margin) : 0,
+      AVSEEK_FLAG_BACKWARD);
+  if (ret < 0) {
+    LOG(WARNING) << "av_seek_frame fails. Stream index: " << index_;
+    return ret;
+  }
+  return 0;
+}
+
+<<<<<<< Updated upstream
```

Comment:
There are some leftover merge conflicts here

Reply:
@brianjo  It is fixed in the replacement PR (https://github.com/pytorch/vision/pull/1303)